{% extends "layout.html" %}

{% block content %}

<div class="jumbotron" style="background:rgba(0,0,0,0.0); border:1px solid black;" >
		<div style="margin-top: -1em;"id='jumbotron_content'>

			<h3> What is DrugThatGene? </h3>
			DrugBuddy is a a web-based application to automate the analysis of potential therapeutic targets and/or therapeutic pathways identified from functional genetic screens. DTG integrates data from human genetic databases and small molecule databases (See 'Databases Utilized by DrugThatGene' heading). A list of "hits" from a pooled CRISPR screen is required in order to use DTG to aid in the identification of drugs/small molecules. 
			
			<h3> Databases Utilized by DrugThatGene </h3>
			Here is the list of databases utilied by DTG:
			<ul>
			<li><a href="http://www.dgidb.org/">The Drug Gene Interaction Database (DGIdb)</a>: Database of drugs with known targets</li>
			<li><a href="https://ocg.cancer.gov/programs/ctd2">Cancer Target Discovery and Development (CTD<sup>2</sup>)</a>: Database of drugs with known targets with a focus on cancer targets</li>
			<li><a href="http://gnomad.broadinstitute.org/">Genome Aggregation Database (gnomAD)</a>: Database aggregating exome and genome sequencing data. The database consists of >120,000 exomes and >15,000 whole  genome sequences.</li>
			<li><a href="https://www.ncbi.nlm.nih.gov/clinvar/">ClinVar</a>: Database of relationships between human variation and phenotypes</li>
			<li><a href="https://www.omim.org/">Online Mendelian Inheritance in Man (OMIM)</a>: A database cataloging human genes and genetic disorders</li>
			<li><a href="http://exac.broadinstitute.org/">Exome Aggregation Consortium (ExAC)</a>: Database of exomes from 60,706 unrelated individuals sequenced as part of various disease-specific and population genetic studies</li>
			<li><a href="https://pharos.nih.gov/idg/index">Pharos</a>: A comprehensive, integrated database for the druggable genome</li>
			<li><a href="https://string-db.org/">STRING</a>: Database of protein-protein interaction networks</li>
			<li><a href="http://mips.helmholtz-muenchen.de/corum/#">CORUM</a>: The comprehensive resource of mammalian protein complexes</li>
			</ul>
			
			<h3> DrugThatGene Table Outputs </h3>
			<b>"Druggable Genes" Tab</b><br>
					Here is a description of each of the table outputs:
					<ul>
					<li>Gene Symbol: Official gene symbol for each gene from the input gene list.</li>
					<li>HGNC ID: HGNC ID for the associated official gene symbol to allow for verification that the correct gene has been identified.</li>
					<li>KEGG Pathways: List of KEGG Pathways for the input gene.</li>
					<li>Complexes: List of protein complexes for the input gene.</li>
					<li>DGIdb #interactions: Number of drug-gene interactions identified within the DGIdb database for the input gene.</li>
					<li>DGIdb interactions: List of drugs/small molecules known to interact with the input gene from the DGIdb database.</li>
					<li>DGIdb: Link to DGIdb for the input gene. See above for a description of the DGIdb database.</li>
					<li>CTD2 interactions: List of drugs/small molecules known to interact with the input gene from the CTD<sup>2</sup> database.</li>
					<li>OMIM: Link to OMIM for the input gene. See above for a description of the OMIM database.</li>
					<li>OMIM variants: Disease associated-variants in the OMIM database</li>
					<li>ClinVar: Link to ClinVar for the input gene. See above for a description of the ClinVar database.</li>
					<li>gnomAD: Link to gnomAD for the input gene. See above for a description of the gnomAD database.</li>
					<li>Pharos: Link to Pharos for the input gene. See above for a description of the Pharos database.</li>
					<li>ExAC: Link to ExAC for the input gene. See above for a description of the ExAC database.</li>
					<li>ExAC #LoF: Number of loss of function (LoF) variants present within the ExAC database.</li>
					<li>ExAC Missense Z: Z score for the deviation of observed counts from the expected number. Positive Z scores indicate increased constraint (intolerance to variation) and therefore that the gene had fewer variants than expected. Negative Z scores are given to genes that had a more variants than expected.</li>
					<li>ExAC pLI: pLI is a score that indicates the probability that a gene is intolerent to a loss of function mutation. The closer pLI is to one, the more LoF intolerant the gene appears to be. pLI >= 0.9 is considered to be an extremely LoF intolerant gene.</li>
					<li>Interaction Map: Protein-protein interaction network from STRING</li>
					
					</ul>
					<b>"Druggable Pathways" Tab</b><br>
					This tab displays data grouped by common pathways. The output table is sorted by the fraction of input genes in each pathway (# of input genes in pathway/total number of genes in pathway)
					<ul>
					<li>KEGG Pathways: Common KEGG Pathways from the input gene list.</li>
					<li>Gene Symbol: Official gene symbol for each gene from the input gene list.</li>
					<li>HGNC ID: HGNC ID for the associated official gene symbol to allow for verification that the correct gene has been identified.</li>
					<li>All Genes in Pathway: List of all genes within the relevant KEGG pathway.</li>
					<li>DGIdb interactions: List of drugs/small molecules known to interact with the input gene from the DGIdb database.</li>
					<li>CTD2 interactions: List of drugs/small molecules known to interact with the input gene from the CTD<sup>2</sup> database.</li>
					<li># of Input Genes in Pathway: The number of genes from the input gene list in the relevant KEGG pathway. The number is parentheses is the fraction of input genes in each pathway (# of input genes in pathway/total number of genes in pathway)</li>
					</ul>
					
					<b>"Druggable Complexes" Tab</b><br>
					This tab displays the data grouped by common protein complexes. The output table is sorted by the fraction of input genes in each protein complex (# of input genes in protein complex/total number of genes in protein complex)
					<ul>
					<li>Complexes: Common CORUM protein complexes from the input gene list.</li>
					<li>Gene Symbol: Official gene symbol for each gene from the input gene list.</li>
					<li>HGNC ID: HGNC ID for the associated official gene symbol to allow for verification that the correct gene has been identified.</li>
					<li>All Genes in Complex: List of all genes within the relevant protein complex.</li>
					<li>Complex Function: Function of the relevant protein complex (determined by FunCat).</li>
					<li>GO Description: Gene ontology (GO) description for the relevant protein complex.</li>
					<li>DGIdb interactions: List of drugs/small molecules known to interact with the input gene from the DGIdb database.</li>
					<li>CTD2 interactions: List of drugs/small molecules known to interact with the input gene from the CTD<sup>2</sup> database.</li>
					<li># of Input Genes in Complex: The number of genes from the input gene list in the relevant protein complex. The number is parentheses is the fraction of input genes in each protein complex (# of input genes in protein complex/total number of genes in protein complex)</li>
					</ul>
					<b>"Missing Genes" Tab</b><br>
					Entered genes that are not Official Gene Symbols (refer to the HGNC database: https://www.genenames.org/) are excluded from analysis. The list of these "missing genes" is presented here.
			
			
			<h3> Performing DTG Analysis </h3>
			Steps to perform DTG analysis: 
			<ol>
			<li>Copy and paste a list of official gene symbols (https://www.genenames.org/) for human genes separated by spaces or on individual lines into the gene list field. The input is case insensitive and duplicate entries will be excluded.</li>
			<li>Enter a number (for example "2", but not "two") to set the number of genes in a pathway required for the software to identify a "common pathway".</li>
			<li>Enter a number (for example "2", but not "two") to set the number of genes in a protein complex required for the software to identify a "common protein complex".</li>
			<li>Click the "Submit" button</li>
			<li>The analysis will be output with four tables: "Druggable Genes", "Druggable Pathways", "Druggable Complexes", and "Missing Genes"</li>
			<li>To download the analyses in .csv format, click "Download Druggable Gene Analysis", "Download Druggable Pathway Analysis", and/or "Download Druggable Complex Analysis"</li>
			<li>Entered genes that are not Official Gene Symbols (refer to the HGNC database: https://www.genenames.org/) are excluded from analysis. To download a list of genes that were excluded from the analysis for this reason, click "Download Genes Missing from Analysis"</li>
			</ol>
			<h3>Example Data</h3>
			Top 100 genes from Riger analysis from genome-wide CRISPR screen in A375 cells (replicate #1):&nbsp <a href="{{ url_for('example_one') }}">Download replicate #1 gene list</a><br>
			Top 100 genes from Riger analysis from genome-wide CRISPR screen in A375 cells (replicate #2):&nbsp <a href="{{ url_for('example_two') }}">Download replicate #2 gene list</a><br> 
			<br>
			Riger analyses are from: Shalem, O., Sanjana, N.E., Hartenian, E., Shi, X., Scott, D.A., Mikkelsen, T.S., Heckl, D., Ebert, B.L., Root, D.E., Doench, J.G., et al. (2014). Genome-scale CRISPR-Cas9 knockout screening in human cells. Science 343, 84â€“87.

			<h3>How to Cite DrugThatGene</h3>
			If you use DrugThatGene in your work, please cite:<br><br>
			Canver, MC., Bauer, DE., Maeda, T., Pinello, L. (2018). Integrative analysis to streamline the identification of druggable genes, pathways, and protein complexes from CRISPR screens. Manuscript in preparation. 
			
			<h3>Acknowledgments</h3>
			DrugThatGene was created by <a href="https://twitter.com/MatthewCanver?lang=en" target="_blank">Matthew C. Canver</a>, <a href="http://www.bauerlab.org/" target="_blank">Daniel E. Bauer</a>, <a href="https://twitter.com/tmhemonc?lang=en" target="_blank">Takahiro Maeda</a> and <a href="http://pinellolab.org/" target="_blank">Luca Pinello</a>
		</div>
	</div>

  
{% endblock %}